ValveResourceFormat/Resource/ResourceTypes/Choreo/Parser/BVCDParser.cs

```diff
@@ -0,0 +1,357 @@
+            var name = ReadString();
+
+            var eventStart = reader.ReadSingle();
+            var eventEnd = reader.ReadSingle();
+
+            var param1 = ReadString();
+            var param2 = ReadString();
+            var param3 = ReadString();
+
+            ChoreoRamp ramp;
+            ramp = ReadRamp();
+
+            if (version >= 17)
+            {
+                //is the first occurance for these at version 17?
+                var unk02 = reader.ReadByte();
+                var unk03 = reader.ReadByte();
+            }
+            var flags = (ChoreoFlags)reader.ReadByte();
+
+            var distanceToTarget = reader.ReadSingle();
+
+            //relative tags
+            var count = reader.ReadByte();
+            var relativeTags = new ChoreoRelativeTag[count];
+            for (var i = 0; i < count; i++)
+            {
+                relativeTags[i] = ReadRelativeTag();
+            }
+
+            //flex timing tags
+            count = reader.ReadByte();
+            var flexTimingTags = new ChoreoFlexTimingTag[count];
+            for (var i = 0; i < count; i++)
+            {
+                flexTimingTags[i] = ReadFlexTimingTag();
+            }
+
+
+            //absolute tags
+            //play tags
+            count = reader.ReadByte();
+            var playTags = new ChoreoAbsoluteTag[count];
+            for (var i = 0; i < count; i++)
+            {
+                playTags[i] = ReadAbsoluteTag();
+            }
+
+            //shift tags
+            count = reader.ReadByte();
+            var shiftTags = new ChoreoAbsoluteTag[count];
+            for (var i = 0; i < count; i++)
+            {
+                shiftTags[i] = ReadAbsoluteTag();
+            }
+
+            var sequenceDuration = -1f;
+            if (eventType == ChoreoEventType.Gesture)
+            {
+                sequenceDuration = reader.ReadSingle();
+            }
+
+            var usingRelativeTag = reader.ReadBoolean();
+            ChoreoRelativeTag relativeTag = null;
+            if (usingRelativeTag)
+            {
+                relativeTag = ReadRelativeTag();
+            }
+
+            var flex = ReadFlex();
+
+            byte loopCount = 0;
+            ChoreoClosedCaptions closedCaptions = null;
+            if (eventType == ChoreoEventType.Loop)
+            {
+                loopCount = reader.ReadByte();
+            }
+            else if (eventType == ChoreoEventType.Speak)
+            {
+                if (version < 17)
+                {
+                    closedCaptions = ReadClosedCaptions();
+                }
+                var unk02 = reader.ReadInt32(); //is this for speak only? what's this? maybe part of closedcaptions?
+                Debug.Assert(unk02 == 0);
+            }
+
+            if (version >= 17)
+            {
+                var unk03 = reader.ReadByte();
+            }
+
+            //eventId or unk01 is sometimes missing?
```

Comment:
After this line it doesn't seem like these bytes belong to choreo event

Reply:
They're definitely there, but I can't see how they get there either.
The server has <16 vcd version.